src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsServiceCollectionExtensions.cs

```diff
@@ -160,11 +160,20 @@ public static IServiceCollection AddApplicationInsights(this IServiceCollection
 
                 var activeConfig = TelemetryConfiguration.Active;
```

Comment:
I see two local instances of the TelemetryConfiguration here;
 - line 150 `TelemetryConfiguration config = TelemetryConfiguration.CreateDefault();`
- line 161 `var activeConfig = TelemetryConfiguration.Active;` 

Note that this call to TelemetryConfiguration.Active is initializing the entire SDK. [see here for reference](https://github.com/microsoft/ApplicationInsights-dotnet/blob/7633ae849edc826a8547745b6bf9f3174715d4bd/BASE/src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs#L108-L125). **I expect this is where your extra traces we discussed below are coming from**

----------
I see that the InstrumentationKey/ConnectionString are set on `activeConfig` and then set again on the `config` variable when it's sent to your `SetupTelemetryConfiguration()` method.
I see from the comments that this is intentional, but this is unintuitive **to me**. 
I would have expected you call to `services.AddSingleton<TelemetryConfiguration>(` to only initialize a TelemetryConfiguration object and not initialize our entire SDK.

I'm not recommending dramatic changes, because obviously this is working for you in production and this is likely out of scope for this PR.
But i'm calling out that this looks suspicious to me.

Reply:
@lmolkova can weigh in on some of the subtleties there -- I know we had some problems that she addressed with setting things both on our Config and the Active config.

Reply:
@TimothyMothra historically we used Active config everywhere and users got used to things like `new TelemetryClient()` and `TC.Active`. It created a lot of problems and, when we can (where dependency injection is available) we create another instance of the configuration. However we do not want to break users who used Active.

So we apply very minimum configuration to Active, just enough for it to allow sending custom telemetry,

With this context, connection strings should be applied to Active AND to the instance of config created by Functions. If we don't do this - we'll end up in the situation, when code works differently in public and non-public clouds (i,e, you can send custom telemetry using Active config in public cloud, but cannot in china-one).